cognee/infrastructure/llm/structured_output_framework/baml/baml_client/runtime.py

```
@@ -99,6 +125,14 @@ async def call_function_async(
         self, *, function_name: str, args: typing.Dict[str, typing.Any]
     ) -> baml_py.baml_py.FunctionResult:
         resolved_options = self.__resolve()
+
+        # Check if already aborted
+        if (
+            resolved_options.abort_controller is not None
+            and resolved_options.abort_controller.aborted
+        ):
+            raise Exception("BamlAbortError: Operation was aborted")
```

Ruff (0.13.1)

Create your own exception

(TRY002)